nilearn/datasets/utils.py

```diff
@@ -527,20 +532,13 @@ def _fetch_file(url, data_dir, resume=True, overwrite=False,
         shutil.move(temp_full_name, full_name)
         dt = time.time() - t0
         if verbose > 0:
-            print('...done. (%i seconds, %i min)' % (dt, dt // 60))
-    except _urllib.error.HTTPError as e:
-        if verbose > 0:
-            print('Error while fetching file %s. Dataset fetching aborted.' %
-                  (file_name))
-        if verbose > 1:
-            print("HTTP Error: %s, %s" % (e, url))
-        raise
-    except _urllib.error.URLError as e:
+            # Complete the reporting hook
+            sys.stderr.write(' ...done. (%i seconds, %i min)\n' % (dt, dt // 60))
+    except (_urllib.error.HTTPError, _urllib.error.URLError) as e:
         if verbose > 0:
-            print('Error while fetching file %s. Dataset fetching aborted.' %
-                  (file_name))
-        if verbose > 1:
-            print("URL Error: %s, %s" % (e, url))
+            e.message = ("Error while fetching file %s; "
```

Comment:
Also just curious, I may have missed it but why not always use a friendly error message rather than only when verbose > 0?